gtsam/hybrid/tests/testDCSAM.cpp

```diff
@@ -0,0 +1,685 @@
+
+  // Use discrete info to get nonlinear factors
+  nfg.push_back(dcMixture.factors()(mostProbableEstimate).first);
+
+  // Setup isam
+  ISAM2Params isam_params;
+  isam_params.relinearizeThreshold = 0.01;
+  isam_params.relinearizeSkip = 1;
+  isam_params.setOptimizationParams(ISAM2DoglegParams());
+  ISAM2 isam(isam_params);
+  isam.update(nfg, values);
+
+  // Solve for updated continuous value
+  values = isam.calculateEstimate();
+
+  // Now update the continuous info in the discrete solver
+  errors = AlgebraicDecisionTree<Key>(dcMixture.factors(), calculateError);
+  dtf = DiscreteFactorFromErrors(dcMixture.discreteKeys(), errors);
+
+  dfg.resize(0);
+  dfg.push_back(dtf);
+
+  // Re-solve discrete to verify that output has switched
+  mostProbableEstimate = dfg.optimize();
+  mpeD = mostProbableEstimate.at(dk.first);
+
+  // Ensure that the prediction is correct
+  EXPECT_LONGS_EQUAL(0, mpeD);
+}
+
+/*
+ * Test full DCSAM solve on DCMixtureFactor for 1D case.
+ *
+ * This is essentially identical to the `DCContinuousFactor` test above,
+ but the
+ * solution is obtained by calling the `DCSAM::update` functions (rather
+ than
+ * implemented manually as above).
+ */
+TEST(DCSAM, SimpleMixtureFactor) {
+  using namespace discrete_mixture_fixture;
+
+  // Make a symbol for a single continuous variable and add to KeyVector
+  KeyVector keys;
+  keys.push_back(x1);
+
+  std::vector<NonlinearFactorValuePair> factorComponents{
+      {f1, prior_noise1->negLogConstant()},
+      {fNullHypo, prior_noiseNullHypo->negLogConstant()}};
+
+  HybridNonlinearFactor dcMixture(dk, factorComponents);
+
+  // Make an empty hybrid factor graph
+  HybridNonlinearFactorGraph hfg;
+
+  hfg.push_back(dcMixture);
+
+  // Let's make an initial guess
+  Values initialGuess;
+  double initVal = -2.5;
+  initialGuess.insert(x1, initVal);
+
+  // We also need an initial guess for the discrete variables
+  // (this will only be used if it is needed by your factors),
+  // here it is ignored.
+  DiscreteValues initialGuessDiscrete;
+  initialGuessDiscrete[dk.first] = 0;
+
+  // Let's make a solver
+  DCSAM dcsam;
+
+  // Add the HybridFactorGraph to DCSAM
+  dcsam.update(hfg, initialGuess, initialGuessDiscrete);
+
+  // Solve
+  HybridValues dcvals = dcsam.calculateEstimate();
+
+  // Run another iteration
+  dcsam.update();
+
+  // Update HybridValues
+  dcvals = dcsam.calculateEstimate();
+
+  // Ensure that the prediction is correct
+  const size_t mpeD = dcvals.discrete().at(dk.first);
+  EXPECT_LONGS_EQUAL(0, mpeD);
+}
+
+namespace slam_fixture {
+const double prior_sigma = 0.1;
+const double meas_sigma = 1.0;
+
+auto prior_noise = noiseModel::Isotropic::Sigma(3, prior_sigma);
+auto meas_noise = noiseModel::Isotropic::Sigma(3, meas_sigma);
+
+Pose2 pose0(0, 0, 0);
+Pose2 dx(1, 0, 0.78539816);
+Pose2 noise(0.01, 0.01, 0.01);
+
+}  // namespace slam_fixture
+
+/**
+ * This is a basic (qualitative) octagonal pose graph SLAM test
```

Comment:
```suggestion
 * This is a basic octagonal pose graph SLAM test
```
It's not qualitative anymore! (Thanks for making this _actually test something_!)